pkg/httpclient/client.go

```diff
@@ -0,0 +1,109 @@
+package httpclient
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+)
+
+// Config is used to modify the HTTPClient behaviour when executing requests.
+type Config struct {
+	RetryPolicy  *RetryPolicy
+	CustomClient *http.Client
+}
+
+// HTTPClient abstracts over the std http.Client and provides a retry mechanism.
+type HTTPClient struct {
+	client *http.Client
+	config Config
+}
+
+// NewHTTPClient returns a new HTTPClient with the provided config.
+// If a CustomClient is provided in the config, the client Timeout will override the RetryPolicy.perRetryTimeout.
+func NewHTTPClient(config Config) *HTTPClient {
```

Comment:
Here we could call it `New`, since the package is named httpclient, so it would be httpclient.New, avoiding stuttering